source/simulator/helper_functions.cc

```diff
@@ -2403,6 +2404,260 @@ namespace aspect
       }
     return new_linear_stokes_solver_tolerance;
   }
+
+
+  template <int dim>
+  void
+  Simulator<dim>::compute_sensitivity_kernel ()
```

Comment:
I'm happy to rename it to what you suggest, but I'm pretty sure based on how sensitivity kernels are defined in the geodynamic literature (including the benchmark) that that's exactly what is calculated here. I actually think calling it the density..._update is less accurate because the update doesn't technically happen here, it happens in the material model, but this is of course directly linked to the compositional fields that are calculated here, so I'm fine with calling it adjoint_update_density_and_viscosity.

Reply:
Maybe `adjoint_update_compositional_fields()` then? Or `adjoint_update_fields_with_gradient()` [of the objective function]? You do an update at the end of the function, after all :-)

Reply:
Thank you for the suggestion. I like adjoint_update_compositional_fields! I changed it to that.